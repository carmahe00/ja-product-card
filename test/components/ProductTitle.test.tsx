import React from 'react'
import renderer from 'react-test-renderer'
import { ProductTitle } from '../../src/components';
import { ProductCard } from '../../src/components/ProductCard';
import { product1, product2 } from '../data/products';
import { ProductImage } from '../../src/components/ProductImage';

describe('ProductTitle', () => {
    test('debe de mostrar el componente correctamente con el titulo personalizado', () =>{
        const wrapper = renderer.create(
            <ProductTitle title="Custom Product" className="custom-class" />
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('debe de mostrar el componente correctamente con el titulo personalizado', () =>{
        const wrapper = renderer.create(
            <ProductCard product={product1} >
                {
                    ()=>(
                        <ProductTitle />
                    )
                }
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })

    test('debe de mostrar el componente correctamente con el titulo personalizado', () =>{
        const wrapper = renderer.create(
            <ProductCard product={product1} >
                {
                    ()=>(
                        <ProductImage img={product2.id} />
                    )
                }
            </ProductCard>
        )
        expect(wrapper.toJSON()).toMatchSnapshot()
    })
})
