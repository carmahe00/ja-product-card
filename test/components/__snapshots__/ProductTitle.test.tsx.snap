// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle debe de mostrar el componente correctamente con el titulo personalizado 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;

exports[`ProductTitle debe de mostrar el componente correctamente con el titulo personalizado 2`] = `
<div
  className="productCard undefined "
>
  <span
    className="productDescription undefined"
  >
    Coffe Mug - Card NoImage
  </span>
</div>
`;

exports[`ProductTitle debe de mostrar el componente correctamente con el titulo personalizado 3`] = `
<div
  className="productCard undefined "
>
  <img
    alt="Producto"
    className="productImg undefined"
    src="2"
  />
</div>
`;
